packages/server/src/api/controllers/static/index.js

```diff
@@ -49,11 +49,9 @@ async function getAppIdFromUrl(ctx) {
     a => a.url && a.url.toLowerCase() === possibleAppUrl
   )[0]
 
-  if (app && app.appId) {
-    return app.appId
-  } else {
-    return ctx.params.appId
-  }
+  const appId = app && app.appId ? app.appId : ctx.params.appId
+  updateAppId(appId)
```

Comment:
I think we should do this from `serveApp` instead, that seems the more appropriate as this is more of a helper, it's also used in migrations so we can't guarantee it's the right thing to do to set context here

Reply:
apologies it's `getAppUrl` that is used in migrations, but the comment above I believe is still valid

Reply:
Since this is a locally scoped and not exported function, I thought it would be best to move the functionality here, as I wouldn't want anyone to try and use the pretty app URL and not understand that the app ID needs to be updated in the context. I felt like that was a more likely scenario than the one of getting the pretty URL but not actually wanting that app ID to be in the context, since this is just to be used in the local scope of serving apps/clients etc.